### DISCUSSION TITLE
what is the diff of  cpu lazy and cpu direct in btop ?

### DISCUSSION THREAD

**Discussion by q2333gh** [aristocratos/btop#672] [General]
I am a fan of OS and btop and a programmer. 
Curious about this .

**Comment by aristocratos** [comment]
@q2333gh 
"Cpu lazy" sorts the processes by cumulative cpu usage over the processes lifetime. But pushes up processes with current cpu usage over a certain threshold to the top (to not miss current top process).

"Cpu direct" sorts only by current cpu usage.

Overall "Cpu lazy" gives a more stable view of the cpu usage since processes will not be as likely to move around as much compared to when sorting by "direct", and to make it easier to track processes when the update timer is set below maybe 2000-3000 ms (which usually is the case).

https://github.com/aristocratos/btop/blob/ad14554f32cc9c19f3cdc8e28ea6012a974a533a/src/btop_shared.cpp#L42-L84

**Comment by q2333gh** [comment]
aha ! `cumulative` what  a good explanation. i got it . thank you!
